File: auth-service/src/main/java/com/nurdorproject/auth_service/dto/RegisterDTO.java
package com.nurdorproject.auth_service.dto;

import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@AllArgsConstructor
@Getter
public class RegisterDTO implements Serializable {
    @NotNull
    @NotBlank
    @Size(max = 100)
    private final String name;

    @NotNull
    @NotBlank
    @Size(max = 100)
    private final String surname;

    @NotNull
    @NotBlank
    @Size(max = 200)
    private final String address;

    @NotNull
    @NotBlank
    @Size(max = 20)
    private final String phoneNumber;

    @NotNull
    @NotBlank
    @Email
    @Size(max = 100)
    private final String email;

    @NotNull
    @NotBlank
    @Size(max = 100)
    private final String username;

    @NotNull
    @NotBlank
    @Size(max = 500)
    private final String password;

    private final byte[] profilePicture;

    @NotNull
    private final String zipCode;

    @NotNull
    @Min(1) @Max(2)
    private final String volunteerRole;
}

File: RMA_Project_v2/app/src/main/java/com/example/rma_nurdor_project_v2/dto/RegisterDTO.kt
package com.example.rma_nurdor_project_v2.dto

data class RegisterDTO(
    val id: Int,
    val name: String,
    val surname: String,
    val address: String,
    val phoneNumber: String,
    val email: String,
    val username: String,
    val password: String,
    val profilePicture: String,
    val nearestCity: String?,
    val volunteerRole: Int?
)
